apps/sms/views/shared/js/activity_picker.js

```diff
@@ -3,87 +3,119 @@
 (function(exports) {
 'use strict';
 
-function handleActivity(activity, onsuccess, onerror) {
-  // Note: the MozActivity is intentionally constructed in the caller and
-  // passed down to this function in order to make it possible to pass an
-  // inline option to its constructor in the caller for the sake of static
-  // analysis tools.  Please do not change that!
+  /**
+   * ActivityPicker
+   *
+   * A collection of some MozActivity methods.
+   */
+  var ActivityPicker = {
 
-  if (typeof onsuccess === 'function') {
-    activity.onsuccess = onsuccess;
-  }
+    /**
+     * When user wants to pass a phone call.
+     * @param  {Number} contact number where to dial.
+     * @returns {Promise} DOMRequest promise is always successfully resolved.
```

Comment:
We don't return the DOMRequest directly, and it's not always resolved successfully. We wrapped it inside Promise.resolve does not mean it could only handle resolve, it's to force it becomes a real promise instance.